CONTRIBUTING.md

```diff
@@ -1,12 +1,141 @@
-- For working on engine - standard `sbt` setup should be enough
-- For working/running GUI please see `ui/README.md` for details instructions
+- For building backend - standard `sbt` setup should be enough
+- For building of frontend `node` and `npm` will be needed - see [client README](ui/client/README.md) for detailed instruction
+- Some tests requires `docker`
+
+### Running
+
+#### Running Designer with embedded model and stubbed DeploymentManager
+
+You can run Nussknacker UI with your model in IDE via
+helper class `LocalNussknackerWithSingleModel`. To use it, add `nussknacker-ui` module to
+test classpath and prepare class similar to `RunDefaultModelLocally`.
+It can be run from e.g. Intellij without special configuration and it will run sample
+Nussknacker UI config with your model.
+
+#### Running full version of Designer from IntelliJ
+
+Before running from IDE you have to manually build:
+- run `npm ci && npm run build` in `ui/client` (only if you want to test/compile FE, see `Readme.md` in `ui/client` for more details)
+- run `prepareDev` in sbt - it prepares components, models and copies FE files (generated above)
+
+Run existing configuration `NussknackerApp` automatically loaded from `./run/NussknackerApp.run.xml`
 
-## Java interoperability
+#### Running full version of Designer from command line
+
+Building:
+- run `npm ci && npm run build` in `ui/client` (only if you want to test/compile FE, see `Readme.md` in `ui/client` for more details)
+- run `./buildServer.sh` in `ui`
+
+Run `./runServer.sh` in `ui`
+
+#### Running using integration environment
+
+- Clone [nussknacker-quickstart](https://github.com/TouK/nussknacker-quickstart)
+- Run `docker-compose -f docker-compose-env.yml -f docker-compose-custom.yml up -d` inside it
+
+#### Accessing service
+
+Service should be available at ~~http://localhost:8080/api~~
+
+#### Troubleshooting
+
+1. If you want to build ui and have access to it from served application, you can execute:
+```
+cd ui/client
+npm ci
+npm run build
+cd ../..
+sbt copyUiDist
+```
+It will produce static assets and copy them to `./ui/server/target/scala-XXX/classes/web/static/` that make them accessible via http://localhost:8080/
+
+```
+cd ui
+cp -r client/.federated-types/nussknackerUi submodules/types/@remote
+cd submodules
+npm ci
+CI=true npm run build
+cd ../..
+sbt copyUiSubmodulesDist
+```
+It will produce submodules static assets and copy them to `./ui/server/target/scala-XXX/classes/web/submodules/` that make them accessible via http://localhost:8080/submodules/*
+
+2. If you want to test verification mechanism, you need to make directory with savepoints available from your dev host. You can use `./bindSavepointsDirLocally.sh` script for that.
+   At the end you need to turn `FLINK_SHOULD_VERIFY_BEFORE_DEPLOY` flag on in environment variables.
+
+### Building
```

Comment:
Let's add a reference to readme in ui/client - I think it's ok that it's in separate file, but reference link would be useful

Reply:
It was in troubleshooting above but good idea - will be more intuitive if it will be in building section. I've added precise steps and reference